rust/integration-tests/src/sdk_tests/mod.rs

```diff
@@ -0,0 +1,138 @@
+use aptos_indexer_testing_framework::{
+    database::{PostgresTestDatabase, TestDatabase},
+    new_test_context::SdkTestContext,
+};
+use assert_json_diff::assert_json_eq;
+use diesel::{Connection, PgConnection};
+use serde_json::Value;
+use std::{collections::HashMap, fs, path::Path};
+
+mod events_processor_tests;
+mod fungible_asset_processor_tests;
+
+#[allow(dead_code)]
+const DEFAULT_OUTPUT_FOLDER: &str = "sdk_expected_db_output_files/";
+
+#[allow(dead_code)]
+pub fn read_and_parse_json(path: &str) -> anyhow::Result<Value> {
+    match fs::read_to_string(path) {
+        Ok(content) => match serde_json::from_str::<Value>(&content) {
+            Ok(json) => Ok(json),
+            Err(e) => {
+                eprintln!("[ERROR] Failed to parse JSON at {}: {}", path, e);
+                Err(anyhow::anyhow!("Failed to parse JSON: {}", e))
+            },
+        },
+        Err(e) => {
+            eprintln!("[ERROR] Failed to read file at {}: {}", path, e);
+            Err(anyhow::anyhow!("Failed to read file: {}", e))
+        },
+    }
+}
+
+// Common setup for database and test context
+#[allow(dead_code)]
+async fn setup_test_environment(transactions: &[&[u8]]) -> (PostgresTestDatabase, SdkTestContext) {
+    let mut db = PostgresTestDatabase::new();
+    db.setup().await.unwrap();
+
+    let test_context = SdkTestContext::new(transactions).await.unwrap();
+
+    (db, test_context)
+}
+
+#[allow(dead_code)]
+fn validate_json(
+    db_values: &mut HashMap<String, Value>,
+    txn_version: u64,
+    processor_name: &str,
+    output_path: String,
+) -> anyhow::Result<()> {
+    for (table_name, db_value) in db_values.iter_mut() {
+        // Generate the expected JSON file path for each table
+        let expected_file_path = Path::new(&output_path)
+            .join(processor_name)
+            .join(txn_version.to_string())
+            .join(format!("{}.json", table_name)); // File name format: processor_table_txnVersion.json
+
+        let mut expected_json = match read_and_parse_json(expected_file_path.to_str().unwrap()) {
+            Ok(json) => json,
+            Err(e) => {
+                eprintln!(
+                    "[ERROR] Error handling JSON for processor {} table {} and transaction version {}: {}",
+                    processor_name, table_name, txn_version, e
+                );
+                panic!("Failed to read and parse JSON for table: {}", table_name);
+            },
+        };
+
+        // TODO: Clean up non-deterministic fields (e.g., timestamps, `inserted_at`)
+        remove_inserted_at(db_value);
+        remove_transaction_timestamp(db_value);
+        remove_inserted_at(&mut expected_json);
+        remove_transaction_timestamp(&mut expected_json);
+
+        assert_json_eq!(db_value, expected_json);
+    }
+    Ok(())
+}
+
+// Helper function to configure and run the processor
+#[allow(dead_code)]
+async fn run_processor_test<F>(
```

Comment:
Is this used anywhere?

Reply:
this is still being used by the existing processor tests. I am trying to separate them out for now for smoother migration.